### ISSUE TITLE
Phase 59.3: PromptTuner — Soft Prompt & Prefix Tuning Framework

### ISSUE THREAD

**Issue by web3guru888** [web3guru888/asi-build#1116]
### Summary
Implement a **PromptTuner** that manages soft prompt tuning, prefix tuning, and P-tuning methods. These approaches prepend learnable continuous embeddings (soft prompts) to the input or hidden states, keeping the entire base model frozen while achieving competitive task performance with extremely few trainable parameters (~0.01% of model size).

#### Soft Prompt Tuning (Lester et al.)
- **Virtual token embeddings**: Learnable soft tokens prepended to input embeddings
- **Prompt length selection**: Configurable prompt length (typically 20–100 tokens)
- **Initialization strategies**: Random, vocabulary-based (from class labels), transferred from another task
- **Prompt ensembling**: Average predictions across multiple prompt initializations

#### Prefix Tuning (Li & Liang)
- **Layer-wise prefixes**: Learnable key-value pairs prepended at every attention layer
- **Reparameterization**: MLP-based prefix generation for training stability
- **Prefix length tuning**: Automatic selection of optimal prefix length per task
- **Prefix sharing**: Shared prefixes across layers with task-specific residuals

#### P-Tuning v2 (Liu et al.)
- **Deep prompt tuning**: Continuous prompts added to every layer (not just input)
- **Anchor tokens**: Interleave trainable and fixed tokens for better optimization
- **Multi-task prompts**: Shared deep prompt backbone with task-specific heads
- **Verbalizer integration**: Map model outputs to task labels via prompt templates

#### Prompt Pool Management
- **Prompt library**: Store and retrieve pre-trained prompts by task/domain
- **Prompt composition**: Concatenate prompts from different tasks for zero-shot transfer
- **Prompt distillation**: Compress long prompts into shorter ones via distillation
- **Instance-aware prompting**: Select/generate prompts conditioned on input features

### API Design
```python
class PromptTuner:
    def create_soft_prompt(self, config: SoftPromptConfig) -> SoftPrompt
    def create_prefix(self, config: PrefixConfig) -> PrefixTuning
    def create_p_tuning(self, config: PTuningConfig) -> PTuningV2
    def initialize_prompt(self, method: str, source: str = "random") -> Tensor
    def compose_prompts(self, prompts: List[SoftPrompt]) -> SoftPrompt
    def distill_prompt(self, long_prompt, target_length: int) -> SoftPrompt
    def manage_prompt_pool(self) -> PromptPool

class SoftPromptConfig:
    num_virtual_tokens: int = 50
    prompt_init: str = "random"  # random, vocab, text, transferred
    init_text: Optional[str] = None
    token_dim: int = 4096
```

### Performance Targets
| Metric | Target |
|--------|--------|
| Trainable parameters | < 0.1% of base model |
| Task accuracy | > 90% of full fine-tuning |
| Prompt transfer success | > 80% on related tasks |
| Prompt distillation retention | > 95% of full prompt quality |
| Training speed | > 3x faster than full fine-tuning |

### References
- Lester et al. (2021) — The Power of Scale for Parameter-Efficient Prompt Tuning
- Li & Liang (2021) — Prefix-Tuning: Optimizing Continuous Prompts for Generation
- Liu et al. (2022) — P-Tuning v2: Prompt Tuning Can Be Comparable to Fine-tuning
- Gu et al. (2022) — PPT: Pre-trained Prompt Tuning for Few-shot Learning
- Wang et al. (2023) — Multitask Prompt Tuning Enables Parameter-Efficient Transfer Learning

### Depends On
- Phase 59.1 (LoRAEngine) for hybrid LoRA+prompt approaches
- Phase 59.5 (PEFTOrchestrator) for unified pipeline

### Part Of
- Phase 59 — Parameter-Efficient Fine-Tuning & Adaptation
- Planning: #1113
